demonstrations/tutorial_testing_symmetry.py

```diff
@@ -0,0 +1,459 @@
+r"""
+Testing for symmetry with quantum computers
+-------------------------------------------
+
+.. meta::
+    :property="og:description": Test if a system possesses discrete symmetries
+    :property="og:image": demonstrations/testing_symmetry/thumbnail.png
+
+*Author: David Wakeham. Posted: 25 November 2021. Last updated: 25 November 2021.*
+
+Introduction
+----------------
```

Comment:
```suggestion
Introduction
------------
```

ReST is a bit picky, it prefers that all heading punctuation marks have the same length as the title

Reply:
I wish I could do everything in markdown 😓